# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/ConcurrentDelazification.h
# Commit: 36780658b33a
# Full Hash: 36780658b33a050e82385161eab2e5e2e5e90064
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2024-12-05 09:58:04
# Regressor Bug: 1933681
# File Overlap Count: 1
# Description:
#   Bug 1933681 - Part 2: Rewrite concurrent delazification's input/output based on InitialStencilAndDelazifications. r=nbp
#   
#   The concurrent delazification internal is still using CompilationStencilMerger.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D230388
# ==============================================================================

diff -r 31d3f2901db9 -r 36780658b33a js/src/vm/ConcurrentDelazification.h
--- a/js/src/vm/ConcurrentDelazification.h	Wed Dec 04 23:44:33 2024 +0000
+++ b/js/src/vm/ConcurrentDelazification.h	Wed Dec 04 23:44:33 2024 +0000
@@ -16,8 +16,9 @@
 #include "frontend/ScriptIndex.h"         // frontend::ScriptIndex
 #include "js/AllocPolicy.h"               // SystemAllocPolicy
 #include "js/CompileOptions.h"  // JS::PrefableCompileOptions, JS::ReadOnlyCompileOptions
-#include "js/UniquePtr.h"  // UniquePtr
-#include "js/Vector.h"     // Vector
+#include "js/experimental/JSStencil.h"  // RefPtrTraits for InitialStencilAndDelazifications
+#include "js/UniquePtr.h"               // UniquePtr
+#include "js/Vector.h"                  // Vector
 
 namespace js {
 
@@ -110,6 +111,8 @@
   // even more functions.
   frontend::CompilationStencilMerger merger_;
 
+  RefPtr<frontend::InitialStencilAndDelazifications> stencils_;
+
   // Record any errors happening while parsing or generating bytecode.
   FrontendContext fc_;
 